drud/ddev
# Add xhprof to ddev, fixes #2974

## The Problem/Issue/Bug:

#2974 Add xhprof 

## How this PR Solves The Problem:

* Adds the xhprof php extension
* Adds command to enable
* Adds the XHProf UI (php) to allow exploring results

Added a bash script as xdebug.

## Manual Testing Instructions:

* Use the [ddev download from this build](https://github.com/drud/ddev/actions/runs/958476279); you can `brew unlink ddev` and just put the downloaded ddev in your PATH. Alternately, use gitpod by visiting [link](https://gitpod.io/#https://github.com/drud/ddev/pull/2983) and you'll have the built item all set up. (You have to be logged into github to see the artifacts)
* If your project has a custom .ddev/apache/apache-site.conf or a custom .ddev/nginx_full/nginx-site.conf, please remove it temporarily (the URL for the xhprof app is added in the default configs)
* `ddev xhprof on`
* Visit a page; Follow (and test) instructions in https://github.com/drud/ddev/blob/8074d8e00a15051385149b873f5b6b0f66b999e7/docs/users/xhprof-profiling.md

## Automated Testing Overview:

* TestDdevXhprofEnabled has pretty much full coverage.

## Related Issue Link(s):

#2974 


<a href="https://gitpod.io/#https://github.com/drud/ddev/pull/2983"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
